### DISCUSSION TITLE
Hyperparameter tunning of DeepAREstimator from gluonTS with Ray

### DISCUSSION THREAD

**Discussion by fissehab** [awslabs/gluonts#1883] [Q&A] [answered]
I want to create forecasting models using the DeepAREstimator from the gluonTS package. How can I use Ray for hyperparameter tuning? Here is sample code.

```
!pip install --upgrade mxnet-cu101==1.6.0.post0
!pip install --upgrade mxnet==1.6.0
!pip install gluonts


import pandas as pd
import matplotlib as mpl
import matplotlib.pyplot as plt
from gluonts.model.deepar import DeepAREstimator
from gluonts.mx.trainer import Trainer
import numpy as np
from gluonts.dataset.common import ListDataset
from gluonts.dataset.field_names import FieldName


#Download data
!wget https://archive.ics.uci.edu/ml/machine-learning-databases/00321/LD2011_2014.txt.zip
!unzip LD2011_2014.txt.zip

df=pd.read_csv('LD2011_2014.txt', sep=';', index_col=0, parse_dates=True, decimal=',')
df_input=df.reset_index(drop=True).T.reset_index()
ts_code=df_input["index"].astype('category').cat.codes.values

#Split to train and test
df_train=df_input.iloc[:,1:134999].values
df_test=df_input.iloc[:,134999:].values

freq="15min"
start_train = pd.Timestamp("2011-01-01 00:15:00", freq=freq)
start_test = pd.Timestamp("2014-11-07 05:30:00", freq=freq)
prediction_lentgh=672
estimator = DeepAREstimator(freq=freq, 
                            context_length=672,
                            prediction_length=prediction_lentgh,
                            use_feat_static_cat=True,
                            cardinality=[1],
                            num_layers=2,
                            num_cells=32,
                            cell_type='lstm',
                            trainer=Trainer(epochs=5))
							
train_ds = ListDataset([
    {
        FieldName.TARGET: target,
        FieldName.START: start_train,
        FieldName.FEAT_STATIC_CAT: fsc
    }
    for (target, fsc) in zip(df_train,
                             ts_code.reshape(-1,1))
], freq=freq)

test_ds = ListDataset([
    {
        FieldName.TARGET: target,
        FieldName.START: start_test,
        FieldName.FEAT_STATIC_CAT: fsc
    }
    for (target, fsc) in zip(df_test,
                            ts_code.reshape(-1,1))
], freq=freq)

predictor = estimator.train(training_data=train_ds)
```

**Comment by geoalgo** [comment, marked as answer]
Hi @fissehab, 

Shameless plug, you can use Syne Tune for this which includes schedulers from Ray (through wrappers) but also other ones (model-based hyperband, multiobjective optimizers, ...), and most importantly we have an example ready for you to tune a gluonts model :-)

https://github.com/awslabs/syne-tune/blob/main/examples/launch_tuning_gluonts.py

**Comment by geoalgo** [comment]
Hi Fisseha,

Sorry for the delay, the message disappeared from the thread, maybe you deleted it?

Local backend and SageMaker refers to trial execution (e.g. how hyperparameters are evaluated).

You can also run Syne Tune in local mode in databrick or elsewhere.

Best,

David


From: Fisseha Gebremariam Berhane ***@***.***>
Reply to: awslabs/gluon-ts ***@***.***>
Date: Wednesday 16 February 2022 at 19:34
To: awslabs/gluon-ts ***@***.***>
Cc: "Salinas, David" ***@***.***>, Mention ***@***.***>
Subject: Re: [awslabs/gluon-ts] Hyperparameter tunning of DeepAREstimator from gluonTS with Ray (Discussion #1883)


In the documentation, it says "(local backend to evaluate them locally; SageMaker to evaluate them as separate SageMaker training jobs; another backend with fast startup times is also in the making)", I guess this means right now it supports SageMaker and local only. Or am I misunderstanding it?

—
Reply to this email directly, view it on GitHub<https://github.com/awslabs/gluon-ts/discussions/1883#discussioncomment-2190868>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ACK5JT4G2ENBFYYIDCKVSYLU3PUWNANCNFSM5OOZSG5Q>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
